app/assets/javascripts/darkswarm/directives/cart_toggle.js.coffee

```diff
@@ -0,0 +1,19 @@
+Darkswarm.directive "cartToggle", ($document) ->
+  # Toggles visibility of the "cart" popover
+  restrict: 'A'
+  link: (scope, elem, attr)->
+    scope.open = false
+
+    $document.bind 'click', (event) ->
+      cart_button = elem[0]
+      element_and_parents = [event.target, event.target.parentElement, event.target.parentElement.parentElement]
+      cart_button_clicked = (element_and_parents.indexOf(cart_button) != -1)
+
+      if cart_button_clicked
+        scope.$apply ->
+          scope.open = !scope.open
+      else
+        scope.$apply ->
+          scope.open = false
+
+      return
```

Comment:
how difficult would it be to unit test this? I ask because I'm not used to stub $document and I don't know if it's tricky or not. I'd test it if possible.

Reply:
I don't think we can test click events on nested DOM elements in `karma`?

Reply:
No idea, tbh